src/pet.js

```diff
@@ -0,0 +1,76 @@
+const MAXIMUM_FITNESS = 10;
+const FITNESS_INCREASE = 4;
+function Pet(name) {
+  this.name = name;
+  this.age = 0;
+  this.hunger = 0;
+  this.fitness = MAXIMUM_FITNESS;
+  this.children = [];
+}
+
+Pet.prototype = {
+  get isAlive() {
+    return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+  },
+};
+
+Pet.prototype.growUp = function () {
+  if (!this.isAlive) {
+    throw new Error("Your pet is no longer alive :(");
+  }
+
+  this.age += 1;
+  this.hunger += 5;
+  this.fitness -= 3;
+};
+
+Pet.prototype.walk = function () {
+  if (!this.isAlive) {
+    throw new Error("Your pet is no longer alive :(");
+  }
+
+  if (this.fitness + FITNESS_INCREASE <= MAXIMUM_FITNESS) {
+    this.fitness += FITNESS_INCREASE;
+  } else {
+    this.fitness = MAXIMUM_FITNESS;
+  }
+};
+
+Pet.prototype.feed = function () {
+  if (!this.isAlive) {
+    throw new Error("Your pet is no longer alive :(");
+  }
+
+  if (this.hunger - 3 >= 0) {
+    this.hunger -= 3;
+  } else {
+    this.hunger = 0;
+  }
+};
+
+Pet.prototype.checkUp = function () {
+  if (this.isAlive) {
+    if (this.fitness <= 3 && this.hunger >= 5) {
+      return "I am hungry AND I need a walk";
+    } else if (this.fitness <= 3) {
+      return "I need a walk";
+    } else if (this.hunger >= 5) {
+      return "I am hungry";
+    } else {
+      return "I feel great!";
+    }
+  } else {
+    return "'Your pet is no longer alive :('";
+  }
```

Comment:
You could clean this up a little bit by not using nested if statements where you don't need to. Its good to be checking if the pet is alive or not at the start, this could be put into its own if statement before the other logic starts in whats called an early exit condition. This means the function will end and not go through any unnecessary code right at the start if that condition is met:
```
  if (!this.isAlive) {
    return "'Your pet is no longer alive :('";
  }
// rest of the code here
```
You can also get rid of the last else statement so there is a default condition should nothing else hit trigger the return up until that point. Putting that together, you get this:
```
  if (!this.isAlive) {
    return "'Your pet is no longer alive :('";
  }

  if (this.fitness <= 3 && this.hunger >= 5) {
    return "I am hungry AND I need a walk";
  } else if (this.fitness <= 3) {
    return "I need a walk";
  } else if (this.hunger >= 5) {
    return "I am hungry";
  }

  return "I feel great!";
```